gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStore.java

```diff
@@ -244,11 +251,12 @@ public boolean releaseQuota(Dag.DagNode<JobExecutionPlan> dagNode) throws IOExce
   }
 
   @Override
-  public Optional<JobStatus> getJobStatus(String flowGroup, String flowName, long flowExecutionId, String jobGroup, String jobName) {
-    Iterator<JobStatus> jobStatusIterator =
-        jobStatusRetriever.getJobStatusesForFlowExecution(flowName, flowGroup, flowExecutionId, jobName, jobGroup);
+  public Optional<JobStatus> getJobStatus(DagNodeId dagNodeId) {
+    Iterator<JobStatus> jobStatusIterator = this.jobStatusRetriever.getJobStatusesForFlowExecution(dagNodeId.getFlowName(),
+        dagNodeId.getFlowGroup(), dagNodeId.getFlowExecutionId(), dagNodeId.getJobName(), dagNodeId.getJobGroup());
 
     if (jobStatusIterator.hasNext()) {
+      // there must exist exactly one job status for a dag node id
```

Comment:
may be worth adding either an assert or at least a `log.error` if this presumption is violated.  what do you think?